GenderPayGap.WebUI/Helpers/ReportFiguresHelper.cs

```diff
@@ -0,0 +1,313 @@
+        }
+        
+        private static void SetOptedOutOfReportingPayQuarterFromDraftReturn(ReportFiguresViewModel viewModel, DraftReturn draftReturn)
+        {
+            viewModel.OptedOutOfReportingPayQuarters = draftReturn.OptedOutOfReportingPayQuarters;
+        }
+        
+        private static void SetOptedOutOfReportingPayQuarterFromSubmittedReturn(ReportFiguresViewModel viewModel, Return submittedReturn)
+        {
+            viewModel.OptedOutOfReportingPayQuarters = submittedReturn.OptedOutOfReportingPayQuarters;
+        }
+        
+        private static void SavePayQuartileFiguresToDraftReturn(ReportFiguresViewModel viewModel, DraftReturn draftReturn)
+        {
+            draftReturn.MaleUpperQuartilePayBand = viewModel.MaleUpperPayBand;
+            draftReturn.FemaleUpperQuartilePayBand = viewModel.FemaleUpperPayBand;
+            draftReturn.MaleUpperPayBand = viewModel.MaleUpperMiddlePayBand;
+            draftReturn.FemaleUpperPayBand = viewModel.FemaleUpperMiddlePayBand;
+            draftReturn.MaleMiddlePayBand = viewModel.MaleLowerMiddlePayBand;
+            draftReturn.FemaleMiddlePayBand = viewModel.FemaleLowerMiddlePayBand;
+            draftReturn.MaleLowerPayBand = viewModel.MaleLowerPayBand;
+            draftReturn.FemaleLowerPayBand = viewModel.FemaleLowerPayBand;
+        }
+        
+        private static void SaveBonusPayFiguresToDraftReturn(ReportFiguresViewModel viewModel, DraftReturn draftReturn)
+        {
+            draftReturn.FemaleMedianBonusPayPercent = viewModel.FemaleBonusPayPercent;
+            draftReturn.MaleMedianBonusPayPercent = viewModel.MaleBonusPayPercent;
+            draftReturn.DiffMeanBonusPercent = viewModel.DiffMeanBonusPercent;
+            draftReturn.DiffMedianBonusPercent = viewModel.DiffMedianBonusPercent;
+        }
+
+        private static void SaveHourlyPayFiguresToDraftReturn(ReportFiguresViewModel viewModel, DraftReturn draftReturn)
+        {
+            draftReturn.DiffMeanHourlyPayPercent = viewModel.DiffMeanHourlyPayPercent;
+            draftReturn.DiffMedianHourlyPercent = viewModel.DiffMedianHourlyPercent;
+        }
+
+        private static void SaveOptedOutOfReportingPayQuartersToDraftReturn(ReportFiguresViewModel viewModel, DraftReturn draftReturn)
+        {
+            draftReturn.OptedOutOfReportingPayQuarters = viewModel.OptedOutOfReportingPayQuarters;
+        }
+        
+        private static void ValidateBonusPayFigures(ReportFiguresViewModel viewModel, HttpRequest request)
+        {
+            viewModel.ParseAndValidateParameters(request, m => m.FemaleBonusPayPercent);
+            viewModel.ParseAndValidateParameters(request, m => m.MaleBonusPayPercent);
+            viewModel.ParseAndValidateParameters(request, m => m.DiffMeanBonusPercent);
+            viewModel.ParseAndValidateParameters(request, m => m.DiffMedianBonusPercent);
+            
+            ValidateBonusPayIntegrity(viewModel);
```

Comment:
if there are already some errors for the mean/median figures we shouldn't check this - we get an error if we try to add an error for the same key